pkg/engines/tekton/utils.go

```diff
@@ -0,0 +1,206 @@
+package tekton
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/jenkins-x/lighthouse/pkg/apis/lighthouse/v1alpha1"
+	"github.com/jenkins-x/lighthouse/pkg/jobutil"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	tektonclient "github.com/tektoncd/pipeline/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilrand "k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	controllerName          = "tekton-controller"
+	gitCloneCatalogTaskName = "git-clone"
+	gitCloneURLParam        = "url"
+	gitCloneRevisionParam   = "revision"
+	gitMergeCatalogTaskName = "git-batch-merge"
+	gitMergeBatchRefsParam  = "batchedRefs"
+)
+
+var reProwExtraRef = regexp.MustCompile(`PROW_EXTRA_GIT_REF_(\d+)`)
+
+// generateBuildID generates a unique build ID for consistency with Prow behavior
+func generateBuildID() string {
+	return fmt.Sprintf("%d", utilrand.Int())
+}
+
+// makePipeline creates a PipelineRun and substitutes LighthouseJob managed pipeline resources with ResourceSpec instead of ResourceRef
+// so that we don't have to take care of potentially dangling created pipeline resources.
+func makePipelineRun(lj v1alpha1.LighthouseJob, logger *logrus.Entry, tektonClient tektonclient.Interface) (*tektonv1beta1.PipelineRun, error) {
+	// First validate.
+	if lj.Spec.PipelineRunSpec == nil {
+		return nil, errors.New("no PipelineSpec defined")
+	}
+
+	buildID := generateBuildID()
+	if buildID == "" {
+		return nil, errors.New("empty BuildID in status")
+	}
+
+	prLabels, annotations := jobutil.LabelsAndAnnotationsForJob(lj, buildID)
+	specCopy := lj.Spec.PipelineRunSpec.DeepCopy()
+	p := tektonv1beta1.PipelineRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations:  annotations,
+			GenerateName: fmt.Sprintf("%s-", lj.Spec.Job),
+			Namespace:    lj.Spec.Namespace,
```

Comment:
Need to not set the namespace here, the target namespace can be different that the lighthouse job namespace.